### ISSUE TITLE
[DataStore] ObserveQuery unexpected behavior when Numeric Index for equality predicate

### ISSUE THREAD

**Issue by chrisbonifacio** [aws-amplify/amplify-js#11018]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

DataStore

### Environment information

<details>

```
  System:
    OS: macOS 13.2
    CPU: (10) arm64 Apple M1 Max
    Memory: 12.36 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.3.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
    Watchman: 2022.12.12.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 110.0.5481.177
    Firefox: 101.0
    Safari: 16.3
  npmPackages:
    @ampproject/toolbox-optimizer:  undefined ()
    @aws-amplify/ui-react: ^4.3.5 => 4.3.5 
    @aws-amplify/ui-react-internal:  undefined ()
    @babel/core:  undefined ()
    @babel/runtime:  7.15.4 
    @edge-runtime/primitives:  2.0.0 
    @hapi/accept:  undefined ()
    @napi-rs/triples:  undefined ()
    @next/font: 13.1.3 => 13.1.3 
    @next/react-dev-overlay:  undefined ()
    @segment/ajv-human-errors:  undefined ()
    @types/node: 18.11.18 => 18.11.18 
    @types/react: 18.0.27 => 18.0.27 
    @types/react-dom: 18.0.10 => 18.0.10 
    @vercel/nft:  undefined ()
    acorn:  undefined ()
    amphtml-validator:  undefined ()
    anser:  undefined ()
    arg:  undefined ()
    assert:  undefined ()
    async-retry:  undefined ()
    async-sema:  undefined ()
    aws-amplify: ^5.0.11 => 5.0.11 
    aws-crt: ^1.15.7 => 1.15.7 
    babel-packages:  undefined ()
    browserify-zlib:  undefined ()
    browserslist:  undefined ()
    buffer:  undefined ()
    bytes:  undefined ()
    chalk:  undefined ()
    ci-info:  undefined ()
    cli-select:  undefined ()
    client-only:  0.0.1 
    comment-json:  undefined ()
    compression:  undefined ()
    conf:  undefined ()
    constants-browserify:  undefined ()
    content-disposition:  undefined ()
    content-type:  undefined ()
    cookie:  undefined ()
    cross-spawn:  undefined ()
    crypto-browserify:  undefined ()
    css.escape:  undefined ()
    cssnano-simple:  undefined ()
    data-uri-to-buffer:  undefined ()
    debug:  undefined ()
    devalue:  undefined ()
    domain-browser:  undefined ()
    edge-runtime:  undefined ()
    encoding: ^0.1.13 => 0.1.13 
    eslint: 8.32.0 => 8.32.0 
    eslint-config-next: 13.1.3 => 13.1.3 
    events:  undefined ()
    find-cache-dir:  undefined ()
    find-up:  undefined ()
    fresh:  undefined ()
    get-orientation:  undefined ()
    glob:  undefined ()
    gzip-size:  undefined ()
    http-proxy:  undefined ()
    https-browserify:  undefined ()
    icss-utils:  undefined ()
    ignore-loader:  undefined ()
    image-size:  undefined ()
    is-animated:  undefined ()
    is-docker:  undefined ()
    is-wsl:  undefined ()
    jest-worker:  undefined ()
    json5:  undefined ()
    jsonwebtoken:  undefined ()
    loader-utils:  undefined ()
    lodash.curry:  undefined ()
    lru-cache:  undefined ()
    micromatch:  undefined ()
    mini-css-extract-plugin:  undefined ()
    nanoid:  undefined ()
    native-url:  undefined ()
    neo-async:  undefined ()
    next: 13.1.3 => 13.1.3 
    node-fetch:  undefined ()
    node-html-parser:  undefined ()
    ora:  undefined ()
    os-browserify:  undefined ()
    p-limit:  undefined ()
    path-browserify:  undefined ()
    platform:  undefined ()
    postcss-flexbugs-fixes:  undefined ()
    postcss-modules-extract-imports:  undefined ()
    postcss-modules-local-by-default:  undefined ()
    postcss-modules-scope:  undefined ()
    postcss-modules-values:  undefined ()
    postcss-preset-env:  undefined ()
    postcss-safe-parser:  undefined ()
    postcss-scss:  undefined ()
    postcss-value-parser:  undefined ()
    process:  undefined ()
    punycode:  undefined ()
    querystring-es3:  undefined ()
    raw-body:  undefined ()
    react: 18.2.0 => 18.2.0 (18.3.0-next-3ba7add60-20221201)
    react-dom: 18.2.0 => 18.2.0 (18.3.0-next-3ba7add60-20221201)
    react-is:  18.2.0 
    react-refresh:  0.12.0 
    react-server-dom-webpack:  undefined ()
    regenerator-runtime:  0.13.4 
    sass-loader:  undefined ()
    scheduler:  undefined ()
    schema-utils:  undefined ()
    semver:  undefined ()
    send:  undefined ()
    server-only:  0.0.1 
    setimmediate:  undefined ()
    shell-quote:  undefined ()
    source-map:  undefined ()
    stacktrace-parser:  undefined ()
    stream-browserify:  undefined ()
    stream-http:  undefined ()
    string-hash:  undefined ()
    string_decoder:  undefined ()
    strip-ansi:  undefined ()
    tar:  undefined ()
    terser:  undefined ()
    text-table:  undefined ()
    timers-browserify:  undefined ()
    tty-browserify:  undefined ()
    typescript: 4.9.4 => 4.9.4 
    ua-parser-js:  undefined ()
    undici:  undefined ()
    unistore:  undefined ()
    util:  undefined ()
    vm-browserify:  undefined ()
    watchpack:  undefined ()
    web-vitals:  undefined ()
    webpack:  undefined ()
    webpack-sources:  undefined ()
    ws:  undefined ()
  npmGlobalPackages:
    @ionic/cli: 6.19.0
    aws-cdk: 2.37.1
    bundle-phobia-cli: 0.14.13
    corepack: 0.10.0
    create-next-app: 13.1.6
    expo-cli: 6.0.5
    npm: 8.3.0
    pm2: 5.1.2
    serve: 13.0.2
    vercel: 23.1.2
    wait-on: 6.0.0
    webpack-bundle-analyzer: 4.6.1
    webpack-cli: 4.10.0
    yarn: 1.22.19


```

</details>

### Describe the bug

Using the following schema:

```graphql
type Post @model @auth(rules: [{ allow: public }]) {
  id: ID!
  title: String!
  content: String @index(name: "byContent")
  comments: [Comment] @hasMany
  rating: Int @index(name: "byRating")
}

type Comment @model @auth(rules: [{ allow: public }]) {
  id: ID!
  content: String!
  post: Post @belongsTo
}
```

When filtering an observeQuery with a numeric index and equality predicate like so:

```js
    const subscription = DataStore.observeQuery(Post, (p) =>
      p.rating.eq(2)
    ).subscribe((snapshot) => {
      const { items, isSynced } = snapshot;
      console.log({ items, isSynced });
    });
```

this results in less records than meet the criteria in the local store, or even no records after refreshing the page

observeQuery data after creating Post:
![Screenshot 2023-02-22 at 4 36 42 PM](https://user-images.githubusercontent.com/16952986/220764918-ddd256e1-22f0-4dc6-bd45-63d8074d406c.png)

after refreshing the page:

![Screenshot 2023-02-22 at 4 36 19 PM](https://user-images.githubusercontent.com/16952986/220764989-9f823d28-6910-4dc7-ba42-a0a85fec6d7a.png)

records in the local store (much more than 3):

![Screenshot 2023-02-22 at 4 40 39 PM](https://user-images.githubusercontent.com/16952986/220765506-52c7aaeb-5fe1-4beb-81b6-7845c4aa881b.png)

### Expected behavior

observeQuery should return the correct amount of records that match the predicate

### Reproduction steps

1. deploy given schema
2. subscribe to Post with predicate on rating field (`p.rating.eq(2)`)
3. refresh page

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by svidgen** [comment]
Taking a look!

**Comment by svidgen** [comment]
The fix (#11019) will be released as part of #11025.

**Comment by svidgen** [comment]
@chrisbonifacio Can you confirm this is fixed for you with [aws-amplify@5.0.16](https://www.npmjs.com/package/aws-amplify/v/5.0.16)? (Same as `@latest` at the time of this comment.)

**Comment by chrisbonifacio** [comment]
@svidgen just upgraded and tested again with the same repro app and can confirm that I'm getting back the correct number of records from the observeQuery on both initial sync and subsequent page refreshes.

![Screen Recording 2023-02-24 at 1 17 12 PM](https://user-images.githubusercontent.com/16952986/221259042-310b04d1-3f2c-45b4-91d5-df85754754ff.gif)
